partner_compassion/models/partner_compassion.py

```diff
@@ -250,3 +277,26 @@ def _can_geocode(self):
         if 'no_geocode' in self.env.context:
             return False
         return super(ResPartner, self)._can_geocode()
+
+    @api.multi
+    def openduplication(self):
+        db_info = self.search([('firstname', '=', self.firstname),
+                               ('lastname', '=', self.lastname),
+                               ('birthdate_date', '=', self.birthdate_date),
+                               ('id', '!=', self.id)])
+        list_hom = []
+        for inf in db_info:
+            list_hom.append((4, inf.id))
```

Comment:
You don't need all this code. Here you can just use `self.partner_duplicate_ids` which was set in the `onchange` method.
Here is a pythonic shorcut :
```
duplicate_list = [(4, id) for id in self.partner_duplicate_ids]
```
Here is useful documentation about [list comprehensions](http://treyhunner.com/2015/12/python-list-comprehensions-now-in-color/)